camus-etl-kafka/src/test/java/com/linkedin/camus/etl/kafka/mapred/EtlInputFormatForUnitTest.java

```diff
@@ -0,0 +1,96 @@
+package com.linkedin.camus.etl.kafka.mapred;
+
+import java.io.IOException;
+
+import kafka.javaapi.TopicMetadataRequest;
+import kafka.javaapi.consumer.SimpleConsumer;
+
+import org.apache.hadoop.mapreduce.InputSplit;
+import org.apache.hadoop.mapreduce.JobContext;
+import org.apache.hadoop.mapreduce.RecordReader;
+import org.apache.hadoop.mapreduce.TaskAttemptContext;
+import org.easymock.EasyMock;
+
+import com.linkedin.camus.coders.CamusWrapper;
+import com.linkedin.camus.etl.kafka.CamusJob;
+import com.linkedin.camus.etl.kafka.common.EtlKey;
+
+
+public class EtlInputFormatForUnitTest extends EtlInputFormat {
+  public static enum ConsumerType {
```

Comment:
using enum to control behavior, e.g., whether we mock consumer, whether we load a different record reader class, etc. If future unit tests need to modify other behaviors of EtlInputFormat, simply add more such enums.